components/automate-deployment/pkg/history/history.go

```diff
@@ -0,0 +1,108 @@
+package history
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/gofrs/uuid"
+	"github.com/sirupsen/logrus"
+)
+
+var nowProvider = func() time.Time {
+	return time.Now()
+}
+
+type Logger struct {
```

Comment:
I'm a little worried about this being confusing since we also have logrus which would be what most people expect from a thing called Logger.

Reply:
Recorder?